py/fiberassign/fba_tertiary_io.py

```diff
@@ -0,0 +1,631 @@
+
+
+def read_targfn(targfn):
+    """
+    Read a tertiary PROGNUM input targets file.
+
+    Args:
+        targfn: full path to the tertiary PROGNUM input targets file (string)
+
+    Returns:
+        targ: a Table() structured array
+        targhdr: the related header
+    """
+    if not os.path.isfile(targfn):
+        msg = "missing {}; exiting".format(targfn)
+        log.error(msg)
+        raise IOerror(msg)
+    targ = Table.read(targfn, "TARGETS")
+    targhdr = targ.meta
+    log.info("reading {} targets from {}".format(len(targ), targfn))
+    return targ, targhdr
+
+
+def read_priofn(priofn):
+    """
+    Read a tertiary PROGNUM input priorities file.
+
+    Args:
+        priofn: full path to the tertiary PROGNUM input priorities file (string)
+
+    Returns:
+        prio:  Table() structured array
+    """
+    if not os.path.isfile(priofn):
+        msg = "missing {}; exiting".format(priofn)
+        log.error(msg)
+        raise IOError(msg)
+    prio = Table.read(priofn)
+    return prio
+
+
+def assert_tertiary_targ(prognum, targfn):
+    """
+    Verifies that a tertiary PROGNUM input targets file is correctly formatted.
+
+    Args:
+        prognum: the tertiary PROGNUM (int)
+        targfn: full path to the tertiary PROGNUM input targets file (string)
+
+    Notes:
+        Will raise an error if the file is not correctly formatted.
+    """
+    # AR read
+    targ, targhdr = read_targfn(targfn)
+
+    # AR check all required keys are there
+    miss_keys = []
+    for key in req_keys:
+        if key not in targ.dtype.names:
+            miss_keys.append(key)
+    if len(miss_keys) > 0:
+        msg = "{} keys are missing from {}; exiting".format(",".join(miss_keys), targfn)
+        log.error(msg)
+        raise IOError(msg)
+
+    # AR check all required header keywords are there
+    miss_hdrkeys = []
+    for hdrkey in req_hdrkeys:
+        if hdrkey not in targhdr:
+            miss_hdrkeys.append(hdrkey)
+    if len(miss_hdrkeys) > 0:
+        msg = "{} keys are missing from the header of {}; exiting".format(
+            ",".join(miss_hdrkeys), targfn
+        )
+        log.error(msg)
+        raise IOError(msg)
+
+    # AR check that the required header keywords are as allowed
+    for hdrkey in allow_hdrkeys:
+        if targhdr[hdrkey] not in allow_hdrkeys[hdrkey]:
+            msg = "hdr['{}']={} not in allowed values ({})".format(
+                hdrkey,
+                targhdr[hdrkey],
+                allow_hdrkeys[hdrkey],
+            )
+            log.error(msg)
+            raise IOError(msg)
+
+    # AR check TARGETID
+    targetids = encode_targetid(
+        release=8888, brickid=prognum, objid=np.arange(len(targ))
+    )
+    sel = targ["TARGETID"] != targetids
+    if sel.sum() > 0:
+        msg = "TARGETID is not as expected, i.e. encode_targetid(release=8888, brickid={}, objid=np.arange(len(targ)))".format(
```

Comment:
For expediency, let's just hardcode `release=8888` at the top for now, but not leave any hanging values of `8888` anywhere else in the code. If somebody ever needs to change the allowed release numbers in the future, it will then at least be easy to track down where `release` is used.

Reply:
ok, so the current code should be fine.
8888 only appears at the top:
https://github.com/desihub/fiberassign/blob/e2f7cfc21729673b66f173f20c23bd59d8a0ae5a/py/fiberassign/fba_tertiary_io.py#L21-L22
(and in one comment)

Reply:
Agreed!